src/services/search/dealerSearch.ts

```diff
@@ -0,0 +1,40 @@
+import { postData, Service } from "../../base"
+
+import { Paginated } from "../../types/pagination"
+import { DealerParams } from "../../types/params/dealer"
+
+import { SearchDealer } from "../../types/models/dealer"
+
+const defaultPagination = {
+  page: 0,
+  size: 3,
+}
+
+const searchForDealers = (path, searchQuery: DealerParams) => {
+  const { pagination, sort, query } = searchQuery
+  const { page, size } = pagination
+
+  const sizeOrDefault =
+    parseInt((size || "").toString(), 10) || defaultPagination.size
+  const pageOrDefault =
+    parseInt((page || "").toString(), 10) - 1 || defaultPagination.page
+
+  const body = {
+    pagination: {
+      page: pageOrDefault,
+      size: sizeOrDefault,
+    },
+    sort: [sort],
+    query,
+  }
+
+  return postData(Service.SEARCH, path, body)
+}
+
+export const fetchDealers = async (
+  query: DealerParams
```

Comment:
can we name this params or similar as it not just contains the query....

Reply:
i don't understand what you mean something like DealerQuery?

Reply:
I ment the name of the param, like it's called query but we are passing and object contingin query pagination and sort... 
Up to you